atc/db/build_in_memory_check.go

```diff
@@ -431,13 +466,21 @@ func (b *inMemoryCheckBuild) PublicPlan() *json.RawMessage {
 	return &m
 }
 
-// ResourceCacheUser return no-user because a check build may only generate a image
-// resource and image resource will be cached by SaveImageResourceVersion.
+// ResourceCacheUser will in-memory build's preId as key in order to avoid unnecessary
+// db init. To ensure preId is unique across all ATCs, also use build's create time in
+// the key.
 func (b *inMemoryCheckBuild) ResourceCacheUser() ResourceCacheUser {
-	return NoUser()
+	return ForInMemoryBuild(b.preId, b.createTime.Nanosecond())
 }
 
+// ContainerOwner use in-memory build id as key. If build id is not generated yet, then
+// call OnCheckBuildStart to do db init. That's cause that, once a container is run, we
+// consider it is doing a real check, and we want to log all container events to db, so
+// that user can realtime view check logs on the GUI.
 func (b *inMemoryCheckBuild) ContainerOwner(planId atc.PlanID) ContainerOwner {
+	if b.id == 0 {
+		b.OnCheckBuildStart()
```

Comment:
I don't think we should call this here - `ContainerOwner` sounds like it should be a "free" call (i.e. doesn't perform any IO). We're also just ignoring the error

Reply:
@aoldershaw This is dirty code that I should have cleaned up.